src/middlewares/ssm.js

```diff
@@ -192,3 +209,34 @@ const invertObject = obj =>
     aggregator[obj[key]] = key
     return aggregator
   }, {})
+
+/**
+ * Convert string to template string and replace variables in paths and name values
+ * @param {Object} Options provided to middleware
+ * @param {Object} Values to be used by the templates
+ */
+const parseStringTemplates = function (options, params) {
+  if (!options.stringTemplates) return options
+  options.paths = Object.keys(options.paths).reduce((result, key) => {
+    result[key] = StringTemplate(options.paths[key], params)
+    return result
+  }, {})
+  options.names = Object.keys(options.names).reduce((result, key) => {
+    result[key] = StringTemplate(options.names[key], params)
+    return result
+  }, {})
+
+  return options
+}
+
+/**
+ * Convert string to template string and replace variables
+ * @param {String} template Template string with variable placeholder allowed by ES6 Template Literals
+ * @param {Object} Values to be used by the template
+ */
+const StringTemplate = function (template, params) {
+  const names = Object.keys(params)
+  const vals = Object.values(params)
+  // eslint-disable-next-line no-new-func
+  return new Function(...names, `return \`${template}\`;`)(...vals)
+}
```

Comment:
This approach is interesting. First time I am seeing it!

Reply:
Only one note in terms of naming: maybe we should call this something like `resolveStringTemplate`.
This should be more explicit. Also since this is not used as a "class" but as a function I would prefer to keep the first letter lowercase.